src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLIER = 2;
+
+    private Node<K, V>[] buckets;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.buckets = new Node[DEFAULT_CAPACITY];
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(getHash(key), key, value, null);
+        if (buckets[getIndex(key)] == null) {
+            buckets[getIndex(key)] = newNode;
+        } else {
+            Node<K, V> lastNode = buckets[getIndex(key)];
+            while (lastNode.next != null || Objects.equals(lastNode.key, newNode.key)) {
+                if (Objects.equals(lastNode.key, newNode.key)) {
+                    lastNode.value = newNode.value;
+                    return;
+                }
+                lastNode = lastNode.next;
+            }
+            lastNode.next = newNode;
+        }
+        size++;
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> checkElement = buckets[getIndex(key)];
+        while (checkElement != null) {
+            if (Objects.equals(checkElement.key, key)) {
+                return checkElement.value;
+            }
+            checkElement = checkElement.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private class Node<K, V> {
+        private int hash;
```

Comment:
Do you need hash field?

Reply:
Generally noooo! Ok I can remove this value!